modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/config/ConfigurationProviderObjectFactory.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.config.dsl.config;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.config.spring.dsl.api.ObjectFactory;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.time.TimeSupplier;
+import org.mule.runtime.extension.api.introspection.config.RuntimeConfigurationModel;
+import org.mule.runtime.extension.api.runtime.ConfigurationProvider;
+import org.mule.runtime.module.extension.internal.config.dsl.AbstractExtensionObjectFactory;
+import org.mule.runtime.module.extension.internal.config.dsl.connection.ConnectionProviderResolver;
+import org.mule.runtime.module.extension.internal.runtime.DynamicConfigPolicy;
+import org.mule.runtime.module.extension.internal.runtime.config.ConfigurationProviderFactory;
+import org.mule.runtime.module.extension.internal.runtime.config.DefaultConfigurationProviderFactory;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ImplicitConnectionProviderValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ResolverSet;
+import org.mule.runtime.module.extension.internal.runtime.resolver.StaticValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ValueResolver;
+
+import java.util.Optional;
+
+/**
+ * A {@link AbstractExtensionObjectFactory} which produces {@link ConfigurationProvider}
+ * instances
+ *
+ * @since 4.0
+ */
+class ConfigurationProviderObjectFactory extends AbstractExtensionObjectFactory<ConfigurationProvider<Object>> implements ObjectFactory<ConfigurationProvider<Object>>
+{
+
+    private final String name;
+    private final RuntimeConfigurationModel configurationModel;
+    private final ConfigurationProviderFactory configurationProviderFactory = new DefaultConfigurationProviderFactory();
+    private final MuleContext muleContext;
+
+    private DynamicConfigPolicy dynamicConfigPolicy;
+    private Optional<ValueResolver<ConnectionProvider>> connectionProviderResolver = empty();
+    private ConfigurationProvider<Object> instance;
+    private boolean requiresConnection = false;
+
+    ConfigurationProviderObjectFactory(String name,
+                                       RuntimeConfigurationModel configurationModel,
+                                       MuleContext muleContext)
+    {
+        this.name = name;
+        this.configurationModel = configurationModel;
+        this.muleContext = muleContext;
+    }
+
+    @Override
+    public ConfigurationProvider<Object> getObject() throws Exception
+    {
+        if (instance == null)
```

Comment:
Try removing this. ObjectFactory will get injection done, then getObject() will be called. And Lifecycle methods will be called over the object created by the ObjectFactory